tracing/tracing.cc

```diff
@@ -138,18 +138,22 @@ trace_state_ptr tracing::create_session(trace_type type, trace_state_props_set p
 }
 
 trace_state_ptr tracing::create_session(const trace_info& secondary_session_info) noexcept {
-    if (!started()) {
-        return nullptr;
-    }
-
     try {
+        bool opentelemetry_tracing = secondary_session_info.state_props.contains(trace_state_props::opentelemetry);
+        if (!started()) {
+            return opentelemetry_tracing ? make_lw_shared<opentelemetry_state>(nullptr, opentelemetry_tracing) : nullptr;
+        }
+
         // Don't create a session if its records are likely to be dropped
         if (!may_create_new_session(secondary_session_info.session_id)) {
```

Comment:
Is this condition still valid? I can imagine that we exceed the limit for "classic tracing sessions", but still want an opentel trace.

Reply:
Ah, sorry, I misunderstood. nvm